src/venue/core.cljs

```diff
@@ -239,13 +246,16 @@
 
 (defn- add-service!
   [{:keys [id handler]}]
-  (if (satisfies? IHandleRequest (handler))
-    (swap! state assoc-in [:services id] handler)
-    (log-severe "Services must implement IHandleRequest." id))
+  (if-not handler
+    (log-severe "Service is null." id)
+    (do
+      (if (satisfies? IHandleRequest (handler))
```

Comment:
similar to above, may not be possible if you really need the logging. not really grokking this so happy to be corrected.